src/libraries/System.Private.CoreLib/src/System/Number.Formatting.cs

```diff
@@ -1608,12 +1623,112 @@ private static unsafe void UInt64ToNumber(ulong value, ref NumberBuffer number)
+        {
+#if TARGET_32BIT
+            while ((uint)(value >> 32) != 0)
+            {
+                bufferEnd = UInt32ToDecChars(bufferEnd, Int64DivMod1E9(ref value), 9);
+            }
+            return UInt32ToDecChars(bufferEnd, (uint)value);
+#else
+            do
+            {
+                ulong remainder;
+                (value, remainder) = Math.DivRem(value, 10);
+                *(--bufferEnd) = (byte)(remainder + '0');
+            }
+            while (value != 0);
+
+            return bufferEnd;
+#endif
+        }
+
+#if TARGET_64BIT
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+#endif
+        internal static unsafe byte* UInt64ToDecChars(byte* bufferEnd, ulong value, int digits)
+        {
+#if TARGET_32BIT
+            while ((uint)(value >> 32) != 0)
+            {
+                bufferEnd = UInt32ToDecChars(bufferEnd, Int64DivMod1E9(ref value), 9);
+                digits -= 9;
+            }
+            return UInt32ToDecChars(bufferEnd, (uint)value, digits);
+#else
+            while (--digits >= 0 || value != 0)
+            {
+                ulong remainder;
+                (value, remainder) = Math.DivRem(value, 10);
+                *(--bufferEnd) = (byte)(remainder + '0');
+            }
+            return bufferEnd;
+#endif
+        }
+
+#if TARGET_64BIT
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+#endif
+        internal static unsafe char* UInt64ToDecChars(char* bufferEnd, ulong value)
+        {
+#if TARGET_32BIT
+            while ((uint)(value >> 32) != 0)
+            {
+                bufferEnd = UInt32ToDecChars(bufferEnd, Int64DivMod1E9(ref value), 9);
+            }
+            return UInt32ToDecChars(bufferEnd, (uint)value);
+#else
+            do
+            {
+                ulong remainder;
+                (value, remainder) = Math.DivRem(value, 10);
+                *(--bufferEnd) = (char)(remainder + '0');
+            }
+            while (value != 0);
+
+            return bufferEnd;
+#endif
+        }
+
+#if TARGET_64BIT
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+#endif
+        internal static unsafe char* UInt64ToDecChars(char* bufferEnd, ulong value, int digits)
+        {
+#if TARGET_32BIT
+            while ((uint)(value >> 32) != 0)
+            {
+                bufferEnd = UInt32ToDecChars(bufferEnd, Int64DivMod1E9(ref value), 9);
+                digits -= 9;
+            }
+            return UInt32ToDecChars(bufferEnd, (uint)value, digits);
+#else
+            while (--digits >= 0 || value != 0)
+            {
+                ulong remainder;
+                (value, remainder) = Math.DivRem(value, 10);
+                *(--bufferEnd) = (char)(remainder + '0');
+            }
+            return bufferEnd;
+#endif
+        }
+
+        internal static unsafe string UInt64ToDecStr(ulong value)
+        {
+            // Intrinsified in mono interpreter
```

Comment:
What does this comment refer to?

Reply:
I believe its referring to `CountDigits`. It's a pre-existing comment that I preserved